src/script.js

```diff
@@ -2,39 +2,66 @@
 
 const SERVER_URL = '{{{SERVER_URL}}}';
 const params = new URLSearchParams(window.location.search);
-const id = params.get('id');
+const videoId = params.get('id');
+let psid;
 
 function handleEvent(data, eventType) {
   const xhr = new XMLHttpRequest();
   xhr.open('POST', SERVER_URL);
   xhr.setRequestHeader('Content-Type', 'application/json');
-  xhr.send(JSON.stringify({ data, eventType }));
+  xhr.send(JSON.stringify({ data, eventType, psid }));
 }
 
-document.addEventListener('DOMContentLoaded', () => {
+function handleError(err, title, message) {
+  const div = document.createElement('div');
+  div.classList.add("error-container");
+  div.innerHTML = `<h1>${title}</h1><p>${message}</p>`;
+  document.querySelector('.container').innerHTML = ``
+  document.querySelector('.container').appendChild(div);
+  console.error(err);
+}
+
+function setPlayer() {
   const options = {
-    id,
-    width: 800
+    id: videoId,
+    responsive: true
   };
+
   const player = new Vimeo.Player('vimeoVideo', options);
 
   player.ready().then(() => {
     player.on('ended', data => handleEvent(data, 'ended'));
```

Comment:
I suspect we should close the webview with MessengerExtensions.requestCloseBrowser when the user has finished the video, no?